src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1291,11 +1291,6 @@ private sealed class LoaderHolder
                 if (loader == null || string.IsNullOrWhiteSpace(loader.Name))
                     goto LDone;
 
-                // Make sure the loader binds to us.
-                var info = host.ComponentCatalog.GetLoadableClassInfo<SignatureDataLoader>(loader.Name);
-                if (info.Type != typeof(ILegacyDataLoader) || info.ArgType != typeof(Options))
```

Comment:
>info [](start = 20, length = 4)

Could we explain this change? As far as I can tell, we're considering not finding this loader in the component factory to be no longer an error condition. This may be right, but I'd feel a bit more comfortable with this change if we can explain why?

I did read the attached issue, but it was a little vague on precisely why removing this check is a right and desirable thing to do.

Reply:
It seems that this check was there in order to make sure that the schema defined in the file has the correct syntax for creating a legacy TextLoader. The schema defined in the file should always be in the format `TextLoader{col=... }`, since it is generated by the `TextSaver`. Instead of the check that I deleted, should we check that `loader.Name==TextLoader.LoaderSignature`?

---
In reply to: [267105243](https://github.com/dotnet/machinelearning/pull/3011#discussion_r267105243) [](ancestors = 267105243)

Reply:
Since it is a text file, might be manually altered after saving it with the TextSaver?

---
In reply to: [267424076](https://github.com/dotnet/machinelearning/pull/3011#discussion_r267424076) [](ancestors = 267424076,267105243)